src/private/context_impl.cpp

```diff
@@ -285,7 +285,13 @@ class Context::ContextImpl {
 		std::unordered_set<std::string> device_list;
 		for (unsigned int i = 0; i < nb_devices; ++i) {
 			auto dev = iio_context_get_device(m_context, i);
-			device_list.emplace(std::string(iio_device_get_name(dev)));
```

Comment:
related to this change: why not do `return Utils::getAllDevices(m_context)` or something like that?

does not have to be in this PR; but it looks doable; unless my coffee did not kick-in yet and i missed something :)

Reply:
Makes sense.
I added the changes.